### ISSUE TITLE
Osgi always restart

### ISSUE THREAD

**Issue by jdotcms** [dotCMS/core#24468]
### Parent Issue

OSGI

### Problem Statement

The extra packages are being parsed on a wrong way, so the OSGI framework always gets restart

### Steps to Reproduce

Upload a plugin twice or more, it always get restart

### Acceptance Criteria

When the same plugin is added, so not restart again and again, b/c the packages have been already added

### dotCMS Version

23.01

### Proposed Objective

Core Features

### Proposed Priority

Priority 2 - Important

**Comment by jdotcms** [comment]
In the past any time you drop a plugin (even if you previously installed), the framework gets restart (you can see the logs on the catalina.out)
Now, after the fix as soon as you installed the first time, the second time you reinstalled, the framework do not get restart

**Comment by jdotcms** [comment]
adding a little bit more context...
use this plugin to test
[Plugin.zip](https://github.com/dotCMS/core/files/11215955/Plugin.zip)

**Comment by jdotcms** [comment]
Unzip the Plugin.zip and D&D the two jars into the Dev Plugin Portlet
First time you will see such as the following logs
```
13:38:26.935  INFO  framework.OSGIUtil - Starting the osgi reload on folder: /Users/jsanca/gitsources/new-core2/tomcat9/webapps/ROOT/WEB-INF/felix/upload
13:38:26.936  INFO  framework.OSGIUtil - OSGI - pathname: language-rule-actionlet-0.2.jar
13:38:26.941  INFO  framework.OSGIUtil - OSGI - pathname: language-rule-actionlet.fragment-0.2.jar
13:38:26.950  INFO  framework.OSGIUtil - There are a new changes into the exported packages
13:38:26.962  INFO  AJAX.OSGIAJAX - OSGI Extra Packages Saved
13:38:30.319  INFO  util.ConcurrentPool - Parsers waiting:20, max at load:20, total created:20, avg creation ms:1606ms @ pool-16-thread-1
13:38:31.969  INFO  concurrent.Debouncer - Debouncing : restartOsgi after 5000 MILLISECONDS
13:38:31.969  INFO  framework.OSGIUtil - Restarting OSGI
13:38:31.971  WARN  util.FileUtil -  Fail to remove destination dir [/Users/jsanca/gitsources/new-core2/tomcat9/webapps/ROOT/WEB-INF/felix/load/language-rule-actionlet-0.2.jar]
13:38:32.022  WARN  util.FileUtil -  Fail to remove destination dir [/Users/jsanca/gitsources/new-core2/tomcat9/webapps/ROOT/WEB-INF/felix/undeployed/language-rule-actionlet.fragment-0.2.jar]
13:38:32.046  INFO  framework.OSGIUtil - Restarting OSGI Cluster Wide
13:38:32.047  INFO  framework.OSGIUtil - Restarting OSGI Locally
13:38:32.047  INFO  framework.OSGIUtil - Portlets Removed: []
13:38:32.060  INFO  framework.OSGIUtil - Felix base dir: /Users/jsanca/gitsources/new-core2/tomcat9/webapps/ROOT/WEB-INF/felix
13:38:32.060  INFO  framework.OSGIUtil - Found property  felix.auto.deploy.action=install,start
13:38:32.060  INFO  framework.OSGIUtil - Found property  felix.fileinstall.log.level=3
13:38:32.060  INFO  framework.OSGIUtil - Found property  felix.org.osgi.framework.startlevel.beginning=2
13:38:32.060  INFO  framework.OSGIUtil - Found property  felix.org.osgi.framework.storage.clean=onFirstInit
13:38:32.060  INFO  framework.OSGIUtil - Found property  felix.log.level=3
13:38:32.060  INFO  framework.OSGIUtil - Found property  felix.cache.locking=false
13:38:32.061  INFO  framework.OSGIUtil - Building Directory:/Users/jsanca/gitsources/new-core2/tomcat9/webapps/ROOT/WEB-INF/felix
13:38:32.061  INFO  framework.OSGIUtil - Building Directory:/Users/jsanca/gitsources/new-core2/tomcat9/webapps/ROOT/WEB-INF/felix/upload
13:38:32.061  INFO  framework.OSGIUtil - Building Directory:/Users/jsanca/gitsources/new-core2/tomcat9/webapps/ROOT/WEB-INF/felix/load
13:38:32.061  INFO  framework.OSGIUtil - Building Directory:/Users/jsanca/gitsources/new-core2/tomcat9/webapps/ROOT/WEB-INF/felix/undeployed
13:38:32.061  INFO  framework.OSGIUtil - Building Directory:/Users/jsanca/gitsources/new-core2/tomcat9/webapps/ROOT/WEB-INF/felix/bundle
13:38:32.061  INFO  framework.OSGIUtil - Building Directory:/Users/jsanca/gitsources/new-core2/tomcat9/webapps/ROOT/WEB-INF/felix/felix-cache
13:38:32.063  INFO  framework.OSGIUtil - Starting the osgi reload on folder: /Users/jsanca/gitsources/new-core2/tomcat9/webapps/ROOT/WEB-INF/felix/upload
13:38:32.063  INFO  framework.OSGIUtil - Loading Factory org.apache.felix.framework.FrameworkFactory
```

**Comment by fmontes** [comment]
When [this action](https://github.com/dotCMS/core/actions/runs/4791962181) finished, you're good to do to test in release docker image.

**Comment by bryanboza** [comment]
Fixed tested on release-23.05 // Docker. 

Now we are not restarting every time you upload the plugin 
```
testing_pg-dotcms-1         | 22:17:17.236  INFO  framework.OSGIUtil - Starting the osgi reload on folder: /data/shared/felix/upload
testing_pg-dotcms-1         | 22:17:17.237  INFO  framework.OSGIUtil - OSGI - pathname: com.dotcms.3rd.party.fragment-0.3.jar
testing_pg-dotcms-1         | 22:17:17.237  INFO  framework.OSGIUtil - OSGI - pathname: com.dotcms.3rd.party-0.3.jar
testing_pg-dotcms-1         | 22:17:21.710  INFO  delegate.SystemEventsJobDelegate - The event: 7e7b3163-2d15-434b-a13d-ac85b95e6364, has been skipped on the server: e93b51ac-974a-42d7-840f-3fec285c6684
testing_pg-dotcms-1         | 22:17:21.710  INFO  delegate.SystemEventsJobDelegate - The event: 25e069da-041d-4887-a159-db2643749e95, has been skipped on the server: e93b51ac-974a-42d7-840f-3fec285c6684
testing_pg-dotcms-1         | 22:17:21.710  INFO  delegate.SystemEventsJobDelegate - The event: b68b4c35-f8e4-466c-a914-650088775c7a, has been skipped on the server: e93b51ac-974a-42d7-840f-3fec285c6684

```
